operator/cmd/operator/app/options/options.go

```diff
@@ -84,6 +87,9 @@ func NewOptions() *Options {
 
 // AddFlags adds flags to the specified FlagSet.
 func (o *Options) AddFlags(fs *pflag.FlagSet, allControllers []string, disabledByDefaultControllers []string) {
+	fs.StringVar(&o.BindAddress, "bind-address", defaultBindAddress, "The IP address on which to listen for the --secure-port port.")
+	fs.IntVar(&o.SecurePort, "secure-port", defaultPort, "The port that the the health server serves at.")
```

Comment:
We don't support TLS verification yet when serving health probe. So this is not a `secure port`.

Do you think we need to introduce port for health probe and metrics separately?

By the way, the `--secure-port` has the same issue.

Reply:
> We don't support TLS verification yet when serving health probe. So this is not a secure port.

how about `admin port`?

> Do you think we need to introduce port for health probe and metrics separately?

whether or not to separate them, I'm ok. but if we name this port `admin port`, I think it's unnecessary to separate them.
if we reach a consensus, there are other components that also have this problem, and we will also need other PRs to solve it.

Reply:
@zhzhuang-zju Please look at this issue. My concern is that the `--secure-port` isn't a real `secure` one.

> if we reach a consensus, there are other components that also have this problem, and we will also need other PRs to solve it.

Yes, exactly.

Reply:
> My concern is that the --secure-port isn't a real secure one.

`--secure-port` isn't a real secure one. refer to https://github.com/karmada-io/karmada/blob/a2e782836d51189242bc1336fa3393adcc0adaa7/operator/cmd/operator/app/options/options.go#L45, there are plans to upgrade to https.

> Do you think we need to introduce port for health probe and metrics separately?

`MetricsBindAddress` contains the port, so there's no need to introduce another port.

> how about admin port?

`admin port` or `sever port` are ok to me

> there are other components that also have this problem, and we will also need other PRs to solve it.

If we want to reach a consensus between different components, we need to sort out the port naming of each component before determining the appropriate portname to avoid conflicts.

Reply:
here are the ports used by all the components I organized

- karmada-operator:
  * `--secure-port` & `--bind-address`: only for http health probe
  * `--metrics-bind-address`: only for metrics
- karmada-agent:
  * `--secure-port` & `--bind-address`: only for http health probe
  * `--metrics-bind-address`: only for metrics
- karmada-controller-manager:
  * `--secure-port` & `--bind-address`: only for http health probe
  * `--metrics-bind-address`: only for metrics
- karmada-descheduler:
  * `--secure-port` & `--bind-address`: for http health probe and metrics
- karmada-scheduler:
  * `--secure-port` & `--bind-address`: for http health probe and metrics
- karmada-scheduler-estimator:
  * `--secure-port` & `--bind-address`: for http health probe and metrics
  * `--server-port`: grpc server port
- karmada-webhook:
  * `--secure-port` & `--bind-address`: only for webhook server
  * `--metrics-bind-address`: only for metrics
  * `--health-probe-bind-address`: only for http health probe

Reply:
From the perspective of consensus among all components, it might be better to separate `health probes` and `metrics`.
- Facilitates subsequent implementation of tls certificate authentication 
  From a data perspective, data from `health probe` is less valuable to do encryption than `metrics`; From the perspective of call frequency, `health probe` is called more frequently and the certificate authentication performance overhead will be higher; Also, the karmada components are mostly `liveness-http` services, so if configure mutual TLS authentication and the `kubelet` doesn't have a certificate issued by karmada, the health check might fail. Refer to https://github.com/kubernetes-sigs/controller-runtime/pull/2407,  controller-runtim is now be possible to serve metrics via tls. So the two have different requirements for tls certificate authentication, in order to facilitate the subsequent implementation of tls certificate authentication, separating the two makes it easier to utilize the extension.
- Consistency with `controller-runtime`
`helath probe` and `metrics` are carried in two separate fields in the controller-runtime's Options, and are not necessarily related to each other.

What do you think? @whitewindmills @RainbowMango

Reply:
I agree with this. 
But another thing I'm not sure about is how what the separated flags look like.  
1. Should we separate the IP address and port?
  * Yes, it might look like:  `--metrics-bind-address` and `--metrics-port`.
  * No, it might look like `--metrics-bind-address`, such as `x.x.x.x:8080`
2. Can the health probes and metrics share the same address and IP?

Reply:
>1. Should we separate the IP address and port?
 
I prefer not to separate ip and port to reduce the number of parameter options to be configured.

> Can the health probes and metrics share the same address and IP?

When choosing to separate `health probes` and `metrics`, we are providing the ability to configure them with different address and ip, which is fine with me if the user configures both as the same.

Reply:
> which is fine with me if the user configures both as the same.

I mean it might be impossible to bind the same port for two HTTPServers...

Reply:
What you concerns are correct. I tried to change the metrics port of `controller-manage` to match the health probe, and then `controller-manager` reported an error
```bash
E0717 07:02:07.663335       1 controllermanager.go:199] controller manager exits unexpectedly: failed to start metrics server: failed to create listener: listen tcp :10357: bind: address already in use
```

Reply:
> What you concerns are correct. I tried to change the metrics port of controller-manage to match the health probe, and then controller-manager reported an error

yes, controller-manager will create two http server for them.

> I mean it might be impossible to bind the same port for two HTTPServers...

usually we don't need to bind the same port for the two http server. and in the production environment, we choose to turn off the metrics .

> No, it might look like --metrics-bind-address, such as x.x.x.x:8080

I vote for this. we just need to be consistent with controller-manager.

Reply:
> I vote for this. we just need to be consistent with controller-manager.

agree

Reply:
+1

Reply:
Looks like we're in agreement :sun_with_face:
@whitewindmills Would you be interested in combing through this task and submitting an issue？

Reply:
SURE, will do that.